src/hooks/use-comment-manger.ts

```diff
@@ -0,0 +1,201 @@
+ *
+ * Software distributed under the License is distributed on an “AS IS” basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the specific language governing rights and limitations under the License.
+ *
+ * The Original Code is manuscripts-frontend.
+ *
+ * The Original Developer is the Initial Developer. The Initial Developer of the Original Code is Atypon Systems LLC.
+ *
+ * All portions of the code written by Atypon Systems LLC are Copyright (c) 2023 Atypon Systems LLC. All Rights Reserved.
+ */
+
+import {
+  deleteHighlightMarkers,
+  isHighlightComment,
+  updateCommentAnnotationState,
+} from '@manuscripts/body-editor'
+import {
+  CommentAnnotation,
+  ElementsOrder,
+  ObjectTypes,
+} from '@manuscripts/json-schema'
+import { buildCommentTree, CommentData } from '@manuscripts/style-guide'
+import { getModelsByType } from '@manuscripts/transform'
+import { EditorView } from 'prosemirror-view'
+import { Dispatch, SetStateAction, useCallback, useMemo } from 'react'
+
+import { useStore } from '../store'
+
+const cleanUpSelectedComment = () => {
+  document
+    .querySelectorAll(`.selected-comment`)
+    .forEach((element) => element.classList.remove('selected-comment'))
+}
+
+/**
+ * Return CRUD callbacks for the comment_list
+ * @param view
+ * @param setSelectedHighlightId
+ */
+export default (
+  view: EditorView | undefined,
+  setSelectedHighlightId: Dispatch<SetStateAction<string | undefined>>
+) => {
+  const [
+    { newComments, doc, trackModelMap, saveTrackModel, deleteTrackModel },
+    dispatch,
+  ] = useStore((store) => ({
+    doc: store.doc,
+    trackModelMap: store.trackModelMap,
+    saveTrackModel: store.saveTrackModel,
+    deleteTrackModel: store.deleteTrackModel,
+    newComments: store.newComments,
+  }))
+
+  const newCommentsList = Array.from(newComments.values())
+
+  /**
+   * Combined new comment(unsaved) with the saved comment.
+   * and add replies for each comment, we do that in *buildCommentTree*
+   */
+  const items = useMemo<Array<[string, CommentData[]]>>(() => {
+    const combinedComments = [
+      ...getModelsByType<CommentAnnotation>(
+        trackModelMap,
+        ObjectTypes.CommentAnnotation
+      ),
+      ...newCommentsList,
+    ]
+    const commentsTreeMap = buildCommentTree(doc, combinedComments)
+    return Array.from(commentsTreeMap.entries())
+  }, [trackModelMap, newCommentsList, doc])
+
+  const removePendingComment = useCallback(
+    (comment: CommentAnnotation) => {
+      if (newComments.has(comment._id)) {
+        newComments.delete(comment._id)
+        dispatch({
+          newComments,
```

Comment:
Here is mutation of a property in the store again. This won't trigger an update for components that take 'newComments' from the store. A new map should be created. E.g.: new Map(newComments)

Reply:
updated